### PR TITLE
EDU-312 Enable batch lesson creation

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-312

### CODE REVIEW

File: src/components/lesson-metadata-modal.yml [hunk 1]
```diff
@@ -10,3 +10,24 @@ titleRequired:
...
+sequenceCount:
+  en: Number
+  de: Anzahl
+sequenceInfoBoxHeader:
+  en: Lesson sequences
+  de: Veranstaltungsreihen
+sequenceInfoBoxDescription:
+  en: |
+    Here you can create multiple lessons at once.
+    The date specified above is used as the first appointment.
```
Comment (adrianaluchian) [inline src/components/lesson-metadata-modal.yml:28]: Maybe "scheduling" would work better than "appointment" here 🤔

File: src/components/lesson-metadata-modal.js [hunk 2]
```diff
@@ -60,12 +78,21 @@ function LessonMetadataModal({ lesson, mode, isVisible, onSave, onClose }) {
     title: lesson.title || t('newLesson'),
     slug: lesson.slug || '',
     language: lesson.language || getDefaultLanguageFromUiLanguage(uiLanguage),
-    startsOn: lesson.schedule?.startsOn ? moment(lesson.schedule?.startsOn) : null
+    startsOn: lesson.schedule?.startsOn ? moment(lesson.schedule?.startsOn) : null,
+    enableSequence: false,
+    sequenceIntervall: LESSON_SEQUENCE_INTERVALL.weekly,
```
Comment (adrianaluchian) [inline src/components/lesson-metadata-modal.js:83]: Some typos here:
- "sequenceIntervall" -> "sequenceInterval"
- "LESSON_SEQUENCE_INTERVALL" -> "LESSON_SEQUENCE_INTERVAL"
- "lessonSequenceIntervallOptions" - "lessonSequenceIntervalOptions"

File: src/components/lesson-metadata-modal.js [hunk 3]
```diff
@@ -156,6 +206,22 @@ function LessonMetadataModal({ lesson, mode, isVisible, onSave, onClose }) {
             showTime={{ defaultValue: moment(`${firstEnabledHour}:00`, 'HH:mm'), format: 'HH:mm', hideDisabledOptions: true }}
             />
         </FormItem>
+        <FormItem name="enableSequence" valuePropName="checked" hidden>
+          <Checkbox />
+        </FormItem>
+        {mode === LESSON_MODAL_MODE.create && (
+          <Collapse className="LessonMetadataModal-sequenceCollapse" activeKey={isSequenceExpanded ? ['sequence'] : []} onChange={handleRepetitionsCollapseChange} ghost>
```
Comment (adrianaluchian) [inline src/components/lesson-metadata-modal.js:213]: Probably an old name here, `handleRepetitionsCollapseChange` -> `handleSequenceCollapseChange`

File: src/components/lesson-metadata-modal.js [hunk 4]
```diff
@@ -156,6 +206,22 @@ function LessonMetadataModal({ lesson, mode, isVisible, onSave, onClose }) {
...
+            <CollapsePanel header={t('createSequence')} key="sequence" forceRender>
+              <Alert className="LessonMetadataModal-sequenceInfo" message={t('sequenceInfoBoxHeader')} description={t('sequenceInfoBoxDescription')} type="info" showIcon />
+              <FormItem label={t('sequenceIntervall')} name="sequenceIntervall" rules={[]}>
```
Comment (adrianaluchian) [inline src/components/lesson-metadata-modal.js:216]: Do `rules` need to be provided at all if there is no validation?

**Comment by adrianaluchian** [comment]
The UI looks super nice 😍

File: src/components/lesson-metadata-modal.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/lesson-metadata-modal.js:213]: Yep, renamed to `handleSequenceCollapseChange`

File: src/components/lesson-metadata-modal.js [hunk 5]
```diff
@@ -126,6 +171,11 @@ function LessonMetadataModal({ lesson, mode, isVisible, onSave, onClose }) {
     };
   };
 
+  const handleRepetitionsCollapseChange = ([value]) => {
```
[the remark this answers is not in the document]
Comment (ahelmberger) [reply src/components/lesson-metadata-modal.js:174]: Ha! It's renamed to `handleSequenceCollapseChange` now anyways :laughing:

File: src/components/lesson-metadata-modal.js (hunk 2 above)
Comment (ahelmberger) [reply src/components/lesson-metadata-modal.js:83]: That was denglisch, now it's english!

File: src/components/lesson-metadata-modal.js [hunk 6]
```diff
@@ -75,7 +102,15 @@ function LessonMetadataModal({ lesson, mode, isVisible, onSave, onClose }) {
     }
   };
 
-  const handleOnFinish = async ({ title, slug, language, startsOn }) => {
+  const handleCancel = () => {
+    onCancel();
+  };
+
+  const handleValuesChange = (_, { startsOn }) => {
+    setHasStartDate(startsOn);
```
[the remark this answers is not in the document]
Comment (ahelmberger) [reply src/components/lesson-metadata-modal.js:110]: Yes, corrected.

File: src/components/lesson-metadata-modal.js (hunk 4 above)
Comment (ahelmberger) [reply src/components/lesson-metadata-modal.js]: No, deleted it here, but I actually provided some rules now for the `sequenceCount` field.

File: src/components/lesson-metadata-modal.yml (hunk 1 above)
Comment (ahelmberger) [reply src/components/lesson-metadata-modal.yml]: Reworded

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**